cmake/dts.cmake

```diff
@@ -88,6 +88,63 @@ if(CONFIG_HAS_DTS)
   # preprocessor, and it seems to be including all kinds of
   # directories with who-knows how many header files.
 
+  # Run the C preprocessor on an empty C source file that has one or
+  # more DTS source files -include'd into it to create the
+  # intermediary file *.dts.pre.tmp
```

Comment:
We need more information about what kind of information we want to get out of the first pass and why the second pass needs the information from the first pass.

Reply:
I'm starting to feel like we need a "Zephyr's hardware model" type document somewhat similar to the multirepo one in https://docs.google.com/document/d/1HrrMZ11nULWoAv3mR70VxT6nB_I1qMytpnmxFcVPCpM/edit?usp=sharing

To clarify: I'm not suggesting we need this to merge this PR, just at some point for the benefit of people's understanding